### DISCUSSION TITLE
Surpressing Instrumentation on a Method-Level

### DISCUSSION THREAD

**Discussion by Taragos** [open-telemetry/opentelemetry-java-instrumentation#5237] [Q&A] [answered]
Hello again,

is it possible to surpress the instrumentation of certain methods?
The use case would be for a Service within Kubernetes or a similiar container orchestration system. Kubernetes performs a health check every few seconds to determine the application status. Currently this health check is also instrumented, so it ends up creating a lot of noise.  

The application in question is a gRPC Netty Service, but I guess the same problem would apply to any other technology that is used and constantly pinged.

Is there a way to surpress these very specific method calls in the javaagent? (I think one possibility would be to filter them out in the Collector they are sent to. But best case I would like to not generate any traces at all for these calls)

**Comment by trask** [comment, marked as answer]
hi @Taragos! check out #1060 for the current available options

**Reply by Taragos** [reply]
Hello @trask,

and thanks for the answer! 

So I guess the best option would be to use [this example](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) as a base to implement a custom sampler with a shouldSample-method that filters based on the name or any other condition one might have?

**Reply by trask** [reply]
yup! we will definitely have configuration based sampling in the future, but first we need a more flexible configuration format, so will take time still. in case it's helpful, also check out https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/samplers/src/main/java/io/opentelemetry/contrib/samplers
